lib/IDE/Refactoring.cpp

```diff
@@ -874,10 +874,23 @@ static void analyzeRenameScope(ValueDecl *VD, Optional<RenameRefInfo> RefInfo,
   }
 
   auto *Scope = VD->getDeclContext();
-  // If the context is a top-level code decl, there may be other sibling
-  // decls that the renamed symbol is visible from
-  if (isa<TopLevelCodeDecl>(Scope))
+  // There may be sibling decls that the renamed symbol is visible from.
+  switch (Scope->getContextKind()) {
+  case DeclContextKind::GenericTypeDecl:
+  case DeclContextKind::ExtensionDecl:
+  case DeclContextKind::TopLevelCodeDecl:
+  case DeclContextKind::SubscriptDecl:
+  case DeclContextKind::EnumElementDecl:
+  case DeclContextKind::AbstractFunctionDecl:
     Scope = Scope->getParent();
+    break;
+  case DeclContextKind::AbstractClosureExpr:
+  case DeclContextKind::Initializer:
+  case DeclContextKind::SerializedLocal:
+  case DeclContextKind::Module:
+  case DeclContextKind::FileUnit:
+    break;
+  }
```

Comment:
I'm not sure this actually goes far enough. I haven't tested this, but I believe this still wouldn't work:
```
func boop() {
  struct Inner {
    enum LocalEnum {
      case east
      case south
    }
  }

  print(Inner.LocalEnum.east)
}
```

The scope would only be `Inner` in that case, so it would still get missed (?). If that is the case, I think we want to go to the last local scope?

Reply:
Good idea. This case wasn’t actually covered. Updated the PR to walk the DeclContext upwards until we reach one that can’t be referenced from the outside.